providers/dns/nearlyfreespeech/internal/client.go

```diff
@@ -32,8 +32,21 @@ type Client struct {
 	HTTPClient *http.Client
 }
 
+// Exposed separately so that tests can share the workaround below.
+func parseURL(uri string) *url.URL {
+	baseURL, err := url.Parse(uri)
+	if err != nil {
+		panic(err)
```

Comment:
panic are allowed

Reply:
I reverted this to silently eating the error.